integration-tests/contracts/ValueCallsWithCompiler.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+// @unsupported: evm
+pragma solidity >=0.7.0;
+
+contract ValueCallsWithCompiler {
+
+    receive() external payable { }
+
+    function getBalance(
```

Comment:
I think this mostly looks good. I would add tests for the following:
* Correct behavior when using delegatecall
* Correct returndata and return status when calling with value but the user does not have enough to transfer
* Correct behavior when not providing enough gas for the newly updated intrinsic gas cost

Reply:
* Need test for self balance